src/main/java/org/springframework/data/elasticsearch/core/AbstractElasticsearchTemplate.java

```diff
@@ -508,7 +529,7 @@ ElasticsearchPersistentEntity<?> getRequiredPersistentEntity(Class<?> clazz) {
 	}
 
 	@Nullable
-	private String getEntityId(Object entity) {
+	public String getEntityId(Object entity) {
 
```

Comment:
That change is not necessary for this issue.

Reply:
It's true this modifycation is unrelated with this PR. However, 

before this PR, I update an entity using spring-data-elasticsearch in this way:
- to get `_id`: ElasticsearchRestTemplate#getEntityId
- to get `_routing`: ElasticsearchRestTemplate#getEntityRouting
- to get `_source`: elasticsearchConverter.mapObject(entity)

(Then combine id/routing/source to a UpdateQuery.)

The latter two methods have public access while the first one is `private` thus **must be copied from `ElasticsearchRestTemplate` to my class** manually to use. If it was `public`, this copy won't happen.

If this PR is merged, seems that there is no need to access `getEntityId` in outer class again. However, `getRouting(entity)` is similar to this method in the same class and be marked as `public`, which is convenient to use. I think it may be better to be accessible for this method too.

Anyway, I'll not modify this line to `public` in new PR, but leave it for your consideration.